carcara/src/parser/mod.rs

```diff
@@ -1200,6 +1327,18 @@ impl<'a, R: BufRead> Parser<'a, R> {
 
                 Ok(result)
             }
+            Token::OpenParen => {
+                self.next_token()?;
+                // ! problem is here -> we might found double parenthesis. (assume a0 (= (concat #b000 ((_ extract 3 3) #b1000))))
+                if self.current_token == Token::ReservedWord(Reserved::Underscore) {
+                    self.next_token()?;
+                }
```

Comment:
I had to add this code to prevent the situation where current_token is an underscore. This is an issue, because `parse_indexed_term` assumes the underscore toekn was already consumed. However, I still feel like this is a hack.